src/index.ts

```diff
@@ -29,173 +26,15 @@ commander
-        status.start({
-          pattern: '  {uptime.green} {spinner.green} Calling validation service...'
-        });
-        let response = await callOmexService(manifest, options);
-        if (response.statusCode === 200) {
-          let formattedBody = JSON.parse(response.body.trim());
-          let validationReport = formattedBody.checkReport.validationReport;
-          let validationResult = validationReport.result;
-          insight.trackEvent('Validation Results', { result: validationResult });
-
-          console.log('-------------------------------------');
-          switch (validationResult) {
-            case 'Passed':
-              // supported products only exist when manifest is valid
-              let supportedProducts = formattedBody.checkReport.details.supportedProducts;
-              console.log(`${chalk.bold('Validation: ')}${chalk.bold.green('Passed')}`);
-              logValidationReport(validationReport.warnings, 'warning');
-              logValidationReport(validationReport.infos, 'info');
-              logSupportedProduct(supportedProducts);
-              break;
-            case 'Failed':
-              console.log(`${chalk.bold('Validation: ')}${chalk.bold.red('Failed')}`);
-              logValidationReport(validationReport.errors, 'error');
-              logValidationReport(validationReport.warnings, 'warning');
-              logValidationReport(validationReport.infos, 'info');
-              break;
-          }
-          console.log('-------------------------------------');
-          process.exitCode = 0;
-        }
-      } catch (err) {
-        if (err.name === 'RequestError') {
-          console.log('-------------------------------------');
-          console.log('Error: Cannot reach service.');
-          console.log('-------------------------------------');
-        } else if (err.name === 'StatusCodeError') {
-          let statusCode = err.statusCode;
-          logError(statusCode);
-        } else {
-          console.log('-------------------------------------');
-          console.log('Error: Unexpected error.');
-          console.log('-------------------------------------');
-        }
-        insight.trackException(new Error('Service Error. Error Code: ' + err.name));
-        // update node process exit code when error is thrown
-        process.exitCode = 1;
-      } finally {
-        // stop progress bar
-        status.stop();
-        process.exit();
-      }
+      process.exitCode = await util.validateManifest(manifest);
+      process.exit;
     } else {
       console.log('-------------------------------------');
       console.log('Error: Please provide a valid local manifest file path.');
       console.log('-------------------------------------');
       insight.trackException(new Error('Manifest file path is not valid.'));
       // update node process exit code when file does not exit
       process.exitCode = 1;
+      process.exit();

```

Comment:
This addresses issue #9, in which the validator doesn't terminate in this code path.